src/main/java/com/shopify/ShopifySdk.java

```diff
@@ -797,21 +878,31 @@ static ObjectMapper buildMapper() {
 
 	private static class ShopifySdkRetryListener implements RetryListener {
 
+		private static final String RETRY_EXCEPTION_ATTEMPT_MESSAGE = "An exception occurred while making an API call to shopify: {} on attempt number {} and {} seconds since first attempt";
+
 		@Override
 		public <V> void onRetry(final Attempt<V> attempt) {
 			if (LOGGER.isWarnEnabled() && attempt.hasResult()) {
 				final Response response = (Response) attempt.getResult();
 				response.bufferEntity();
 				if (shouldRetryResponse(response) && !hasExceededRateLimit(response)) {
-					final long delaySinceFirstAttemptInMilliseconds = attempt.getDelaySinceFirstAttempt();
-					final long delaySinceFirstAttemptInSeconds = TimeUnit.SECONDS
-							.convert(delaySinceFirstAttemptInMilliseconds, TimeUnit.MILLISECONDS);
-					LOGGER.warn(RETRY_ATTEMPT_MESSAGE, delaySinceFirstAttemptInSeconds, attempt.getAttemptNumber(),
-							response.getLocation(), response.getStatus(), response.getStringHeaders(),
-							response.readEntity(String.class));
+					final long delaySinceFirstAttemptInSeconds = convertMillisecondsToSeconds(
+							attempt.getDelaySinceFirstAttempt());
+					LOGGER.warn(RETRY_INVALID_RESPONSE_ATTEMPT_MESSAGE, delaySinceFirstAttemptInSeconds,
+							attempt.getAttemptNumber(), response.getLocation(), response.getStatus(),
+							response.getStringHeaders(), response.readEntity(String.class));
 				}
+			} else if (LOGGER.isWarnEnabled() && attempt.hasException()) {
+				final long delaySinceFirstAttemptInSeconds = convertMillisecondsToSeconds(
+						attempt.getDelaySinceFirstAttempt());
+				LOGGER.warn(RETRY_EXCEPTION_ATTEMPT_MESSAGE, attempt.getExceptionCause(), attempt.getAttemptNumber(),
```

Comment:
Please make the exception the last arg here since it will most likely spill onto a new lines with the stack trace.

Reply:
@rjdavis3 the guava retryer only gives you the the exception cause from what I can see.

Reply:
Oh nevermind i see what you mean. It was just the exception name, but i will move it to the last arg here.